Snowflake.Data.Tests/UnitTests/Configuration/EasyLoggingConfigProviderTest.cs

```diff
@@ -0,0 +1,54 @@
+using Moq;
+using NUnit.Framework;
+using Snowflake.Data.Configuration;
+
+namespace Snowflake.Data.Tests.UnitTests.Configuration
+{
+    [TestFixture]
+    public class EasyLoggingConfigProviderTest
+    {
+        private const string FilePathFromConnectionString = "/Users/dotnet/config.json";
+        private const string FilePathToUse = "/home/config.json";
+        
+        [Test]
+        public void TestThatProvidesConfiguration()
+        {
+            // arrange
+            var configFinder = new Mock<EasyLoggingConfigFinder>();
+            var configParser = new Mock<EasyLoggingConfigParser>();
+            var configProvider = new EasyLoggingConfigProvider(configFinder.Object, configParser.Object);
+            var config = new EasyLoggingConfig();
+            configFinder
+                .Setup(finder => finder.FindConfigFilePath(FilePathFromConnectionString))
+                .Returns(FilePathToUse);
+            configParser
+                .Setup(parser => parser.Parse(FilePathToUse))
+                .Returns(config);
+
+            // act
+            var result = configProvider.ProvideConfig(FilePathFromConnectionString);
+            
+            // assert
+            Assert.AreSame(config, result);
+        }
+
+        [Test]
+        public void TestThatReturnsNullWhenNoConfigurationFound()
+        {
+            // arrange
+            var configFinder = new Mock<EasyLoggingConfigFinder>();
+            var configParser = new Mock<EasyLoggingConfigParser>();
+            var configProvider = new EasyLoggingConfigProvider(configFinder.Object, configParser.Object);
+            configFinder
+                .Setup(finder => finder.FindConfigFilePath(FilePathFromConnectionString))
+                .Returns((string) null);
+            
+            // act
+            var result = configProvider.ProvideConfig(FilePathFromConnectionString);
+            
+            // assert
+            Assert.IsNull(result);
+            configParser.Verify(parser => parser.Parse(It.IsAny<string>()), Times.Never);
```

Comment:
I'm not sure if it sould be a part of the test.
But if so, maybe add Assert.DoesNotThrow because Verify may throw an exception

Reply:
this Verify is to test that we do not try to parse file if we didn't find any. Maybe this verify is not so important.
However I don't understand the comment about using Assert.DoesNotThrow. If Verify would throw an exception - that's perfectly fine - it would indicate that the code does not work as expected.

Reply:
I meant something like this:
Assert.DoesNotThrow(configParser.Verify(parser => parser.Parse(It.IsAny<string>()), Times.Never));

But up to you.